torch/quantization/fx/prepare.py

```diff
@@ -940,6 +948,24 @@ def insert_observers_for_model(
             if not skip_inserting_observers:
                 modules = dict(model.named_modules(remove_duplicate=False))
                 if node.op != 'output':
+
+                    # This is currently only used for equalization.
+                    # Checks if the current node is in a branch by seeing if its
+                    # previous node has more than 1 user.
+                    #
+                    # ex.       conv2
+                    #         /
+                    #      x -> conv1
+                    #
+                    # conv1 and conv2 are both in branches because its previous
+                    # node (x) has 2 users (conv1 and conv2). These layers will
+                    # currently not get equalized.
+                    is_branch = (
```

Comment:
Can we move this check inside `maybe_insert_input_equalization_observers_for_node`?

Reply:
I wanted to keep the check inside `maybe_insert_input_equalization_observers_for_node`, but the reason why I had it before was because the previous `maybe_insert_input_observers_for_node` modifies the graph by inserting quantization observers.

For example:
```
     conv2 
    /     \
x -> conv1 -> add
```

Looking at conv1, insert_input_observers() will insert a quantization observer and the graph will be modified as follows:
```
             conv2 
    /                       \
x -> inp_quant_obs -> conv1 -> add
```
So then when we call insert_equalization_observers(), we would need to check 2 nodes before conv1 to see if it has 2 users and is in a branch.

But when we are looking at conv2, insert_input_observers() will modify the graph to be the following:
```
                       conv2 
                      /     \
x -> inp_quant_obs -> conv1 -> add
```
In this case we could check the 1 node before conv2 and see if it has 2 users to see if conv2 is in a branch.

So, the loops would need two different casings, which is why I decided to add the check for whether the node is in a branch before the insert_input_observer() function. Does that make sense?